src/ut_control_plane.c

```diff
@@ -117,7 +117,19 @@ static cp_message_t* dequeue_message(ut_cp_instance_internal_t *pInternal)
     {
         pthread_cond_wait(&pInternal->queue_condition, &pInternal->queue_mutex);
     }
-    msg = &pInternal->message_queue[0];
+    //msg = &pInternal->message_queue[0];
+    msg = (cp_message_t*)malloc(sizeof(cp_message_t));
+    if (msg == NULL)
+    {
+        pthread_mutex_unlock(&pInternal->queue_mutex);
+        return NULL; // Return NULL if memory allocation fails
+    }
+
+    // Copy the message content to the newly allocated memory
+    msg->size = pInternal->message_queue[0].size;
```

Comment:
It's not a major issue, but you should be aware of this.

1. **Thread A (Producer)**: 
   - Acquires the mutex.
   - Enqueues a message if there's space.
   - Signals the condition variable, then releases the mutex.

2. **Thread B (Consumer)**:
   - Wakes up due to the signal.
   - Attempts to acquire the mutex. If Thread A still holds it, Thread B will block.
   - Once the mutex is available, Thread B acquires it, checks the condition (message count), processes messages, then releases the mutex.